File: apple/LC88MergeSortedArray.java
class Solution {
    public void merge(int[] nums1, int m, int[] nums2, int n) {
        int t1 = m-1;
        int t2 = n-1;
        int total = m+n-1;
        while (t1 >= 0 && t2 >= 0) {
            nums1[total--] = nums1[t1] > nums2[t2] ? nums1[t1--] : nums2[t2--];
        }
        while (t2 >= 0) {
            nums1[total--] = nums2[t2--];
        }
    }
}

File: EA/LC88MergeSortedArray.py
class Solution(object):
    def merge(self, nums1, m, nums2, n):
        temp = m+n-1
        i = m-1
        j = n-1
        while i >= 0 and j >= 0:
            if nums1[i] >= nums2[j]:
                nums1[temp] = nums1[i]
                temp -= 1
                i -= 1
            else:
                nums1[temp] = nums2[j]
                j -= 1
                temp -= 1
        while j >= 0:
            nums1[temp] = nums2[j]
            temp -= 1
            j -= 1
        return
